tools/gfx-unit-test/shared-textures-tests.cpp

```diff
@@ -114,7 +114,7 @@ namespace gfx_test
             ResourceState::CopyDestination,
             ResourceState::CopySource);
         bufferDesc.defaultState = ResourceState::UnorderedAccess;
-        bufferDesc.cpuAccessFlags = MemoryType::GpuOnly;
+        bufferDesc.memoryType = MemoryType::DeviceLocal;
```

Comment:
MemoryType is just to be specific about what heap/memory to allocate the resource from. Since gfx don't do automatic versioning, it is simpler to let user decide what memory type it is and live with the constraints of that memory type.

Reply:
I think the interesting question here is whether we want `gfx` to:

* Expose the bare mechanism ("here's a list of places where current GPUs and GPU APIs support allocating resources; pick one"), and then have to deal with the various cases where we have a back-end that doesn't match the assumptions being made and we have to hand-wave or cut corners (which `memoryType` corresponds to memory allocated with `cudaMallocManaged` vs. `cudaMalloc` for our CUDA back-end?)

* Expose an explicit list of policies which are phrased in terms of the application's intention, and which have both specific semantic constraints/guarantees (e.g., "this policy means the resource must not be read from CPU code", ...) and also certain guarantees about how they map on particular back-ends (e.g., "for the D3D12 back-end, this policy maps to the such-and-such heap")

At some level, this is basically just a matter of what we name the `enum` and its cases, but I tend to favor the second style insofar as we have goals for fairly broad portability, and part of how we ended up in the situation of having to make this entire API/layer is because it turns out that there are APIs and target platforms that don't conform to the model that D3D12 and Vulkan want to assume.